components/layout/construct.rs

```diff
@@ -750,13 +750,13 @@ impl<'a, ConcreteThreadSafeLayoutNode: ThreadSafeLayoutNode>
             }
             TextContent::GeneratedContent(content_items) => {
                 for content_item in content_items.into_iter() {
-                    let specific_fragment_info = match content_item {
-                        ContentItem::String(string) => {
-                            let info = Box::new(UnscannedTextFragmentInfo::new(string, None));
+                    let specific_fragment_info = match *content_item {
+                        ContentItem::String(ref string) => {
+                            let info = Box::new(UnscannedTextFragmentInfo::new(String::from(string.clone()), None));
```

Comment:
Could we change the implementation of `UnscannedTextFragmentInfo` to take a `Box<str>` instead of a `String`? This conversion seems a bit redundant.
https://github.com/servo/servo/blob/8f24005e17aa0b5e274e6efbe0e44f208eb71e42/components/layout/fragment.rs#L570-L575